graylog2-server/src/main/java/org/graylog2/web/resources/AppConfigResource.java

```diff
@@ -68,7 +80,16 @@ public String get(@Context HttpHeaders headers) {
         final Map<String, Object> model = ImmutableMap.of(
             "rootTimeZone", configuration.getRootTimeZone(),
             "serverUri", baseUri.resolve(HttpConfiguration.PATH_API),
-            "appPathPrefix", baseUri.getPath());
+            "appPathPrefix", baseUri.getPath(),
+            "pluginUISettings", buildPluginUISettings());
         return templateEngine.transform(template, model);
     }
+
+    private String buildPluginUISettings() {
+        try {
+            return objectMapper.writeValueAsString(pluginUISettings);
```

Comment:
This will write the `pluginSettingsJson`, which is a JSON string, as JSON string again and thus escaping it. Is that intended? Should we use a `Map<String, Object>` instead of a JSON string as `PluginUISettingsProvider`?

Reply:
Yeah, I will work on a way to make this both more flexible and more serializable

Reply:
We have to do the JSON string conversion before passing data to the template engine because the template engine will do the conversion properly.